### PR TITLE
Feature/dock 1765/categories summary buttons

### PR DESCRIPTION
**Pull request by svonworl**
**Description**
This PR is the second of three PRs constituting the initial implementation of the Categories UI.  It implements the "Browse popular categories" feature on the login page, and the categories buttons on each entry summary on the collection page.

The "Browse popular categories" feature is a fairly faithful implementation of the UI mockup:
https://app.zeplin.io/project/5fc690a2717ee9144814f190
To see it, create a few categories, add some entries to them, then log out and go to the home page.  Our "logged-in" home page is so much different than the mockups that I didn't add it there, because I wasn't sure if/how/where/in-what-style to do so.  Guidance, please!

Currently, "popular" is defined as the categories containing the most workflows/tools, per Denis's suggestion.  Up to four categories of each type are displayed.  The same category may appear on both the workflow and tool side, if it contains both (this behavior is present in the mockup).  Currently, all categories are retrieved from the server, then filtered.  Someday, when we have lots of categories, and we have a better idea of what we want "popular" to mean, we'll probably want to create/modify an endpoint to return the popular categories, exclusively.

Per the Material recommendations, the little round rectangles that denote each category are anchor mat-buttons, because they navigate to another page:
https://material.angular.io/components/button/overview
They are implemented as the app-category-button component.

Each button is labeled with the category's displayName and has a tooltip that displays the topic.  Per Charles' input, each button links to the search page for the category.  I highly recommend that we create and link to dedicated category pages in a follow-on PR, it will vastly improve the experience.

What's left?
1. Add the category buttons to the entry page.
2. Add the category buttons to the logged-in home page, if desired.
3. Write tests.  I didn't write any tests for this PR, I would like to write and include them with the subsequent PR, so there's less risk of needing to rewrite them.

My Angular grok level is much higher than a week ago, so all that should go relatively quickly.

**Screenshots**
Screen Shot 2021-11-08 at 2 59 31 PM

Screen Shot 2021-11-08 at 3 00 40 PM

**Issue**
dockstore/dockstore#4476
https://ucsc-cgl.atlassian.net/browse/DOCK-1765

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that your code compiles by running `npm run build`
- [x] If this is the first time you're submitting a PR or even if you just need a refresher, consider reviewing our [style guide](https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#pr-checklist)
- [x] Do not bypass Angular sanitization (bypassSecurityTrustHtml, etc.), or justify why you need to do so
- [x] If displaying markdown, use the `markdown-wrapper` component, which does extra sanitization
- [x] Do not use cookies, although this may change in the future
- [ ] Run `npm audit` and ensure you are not introducing new vulnerabilities
- [x] Do due diligence on new 3rd party libraries, checking for CVEs
- [x] Don't allow user-uploaded images to be served from the Dockstore domain

### CODE REVIEW

**Comment by coverbeck** [comment]
Can you please attach screenshot(s) of your changes.

**Comment by svonworl** [comment]
> Can you please attach screenshot(s) of your changes.

Oops, I forgot to click the "Update" button.  Done!

**Comment by denis-yuen** [comment]
Screenshots look good and match the mock-ups

>  To see it, create a few categories, add some entries to them, then log out and go to the home page. Our "logged-in" home page is so much different than the mockups that I didn't add it there, because I wasn't sure if/how/where/in-what-style to do so. Guidance, please! 

FWIW, I don't actually see any proposed categories on the Dashboard mock-ups.

**Review by coverbeck — APPROVED** [review]
>  I would like to write and include them with the subsequent PR, so there's less risk of needing to rewrite them.

Mixed feelings about this, but OK.

**Comment by svonworl** [comment]
> Screenshots look good and match the mock-ups
> 
> > To see it, create a few categories, add some entries to them, then log out and go to the home page. Our "logged-in" home page is so much different than the mockups that I didn't add it there, because I wasn't sure if/how/where/in-what-style to do so. Guidance, please!
> 
> FWIW, I don't actually see any proposed categories on the Dashboard mock-ups.

Oh, there's this mockup of the logged-in homepage:
https://app.zeplin.io/project/5fc690a2717ee9144814f190/screen/602198cd52ff99146b777af4
I see the Dashboard mocks below, now I know the current "dashboardy" logged-in homepage UI is intentional. :)

**Comment by denis-yuen** [comment]
Oh, the homepage. I must have misread dashboard, thanks

File: src/app/home-page/home-logged-out/home.component.html [hunk 1]
```diff
@@ -60,223 +60,237 @@ <h2>An app store for bioinformatics</h2>
...
-          </div>
-          <img src="../../../assets/images/home/diagram-launch-with.png" alt="diagram" fxHide.lt-md fxFlex="45" class="contain-img" />
-        </div>
-      </mat-tab>
-      <mat-tab>
-        <ng-template mat-tab-label>
-          <div fxLayoutGap="0.5rem">
-            <img src="../../../assets/images/home/icons-navigation-about.svg" alt="cloud icon" />
-            <span>Use Locally</span>
+  <div *ngIf="(toolCategories$ | async).length + (workflowCategories$ | async).length > 0" fxLayout="column" class="categories">
```
Comment (garyluu) [inline src/app/home-page/home-logged-out/home.component.html:63]: what happens without the fxlayout?

**Review by garyluu — COMMENTED** [review]
Overall, remove unused variables and follow the PR checklist https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#pr-checklist

File: src/app/home-page/home-logged-out/home.component.html (hunk 1 above)
Comment (svonworl) [reply src/app/home-page/home-logged-out/home.component.html]: Looks like leftover cruft, doesn't do anything, removed.

**Review by garyluu — APPROVED** [review]

**Review by denis-yuen — APPROVED** [review]

**Pull request merged**